### ISSUE TITLE
Sliding Sync does not seem to work

### ISSUE THREAD

**Issue by erebion** [element-hq/synapse#17651]
### Description

Release notes mention that Synapse 1.114.0 enabled Sliding Sync by default:

>Enable native sliding sync support (https://github.com/matrix-org/matrix-spec-proposals/pull/3575 and https://github.com/matrix-org/matrix-spec-proposals/pull/4186) by default. (https://github.com/element-hq/synapse/issues/17648)

I've upgraded Synapse and tried logging in with Element X.

It just tells me something like "This homeserver does not currently support SlidingSync.".

### Steps to reproduce

- ugrade to 1.114.0
- try to log in using Sliding Sync
- Element X claims Sliding Sync is not suported.

### Homeserver

erebion.eu

### Synapse Version

1.114.0

### Installation Method

Docker (matrixdotorg/synapse)

### Database

Postgres, single container

### Workers

Single process

### Platform

Debian Stable

### Relevant log output

```shell
There's nothing in the logs that I can find.
```

**Comment by YamatoRyou** [comment]
"Sliding Sync is enabled by default in Synapse 1.114.0" refers to "Simplified Sliding Sync"
Native Sliding Sync is disabled by default in Elemnet X (0.5.1). Your server needs to support Sliding Sync (proxy) in order to switch to native Sliding Sync later. This will cause your server to use both Sliding Sync implementations at the same time.

1. Visit https://github.com/matrix-org/sliding-sync and follow the instructions to run the Sliding Sync (proxy implementation) instance;
2. Log in to your server using Element X;
3. On the Element X settings page, click the version number 7 times in a row, then go to "Developer Options";
4. Find and enable "Simplified Sliding Sync";
5. Log in to your server again.

----
If you have a way to access the developer options before logging in, just do step 5.

**Comment by immanuelfodor** [comment]
It reads like the worst possible UX. You first need to have the Sliding Sync Proxy for not depending on it then?! :sweat_smile: I wanted to avoid installing the sliding sync proxy in the first place.

**Comment by erikjohnston** [comment]
Note that currently released versions of Element X do need the proxy, this is more to support the upcoming releases which won't require them. We may have jumped the gun slightly by sticking it at the top of the changelog!

**Comment by avsm** [comment]
Thanks for tha clarification @erikjohnston! I ran into the exact same problem as @immanuelfodor -- I'm excited to try Element X, but also can't install the sliding sync proxy.  I guess I'll keep an eye out for future updates of Element X that will work without the proxy, now that my server's updated...

**Comment by SNThrailkill** [comment]
I was so excited to try out sliding sync and then I saw this issue. Is this now just waiting on the Element X side or is there more needed in the Synapse side like for auth?

**Comment by paboum** [comment]
Okay, so when is it going to be actually **released** and **supported**?

As in:
  * https://element-hq.github.io/synapse/latest/usage/configuration/config_documentation.html will describe the config
  * https://github.com/matrix-org/sliding-sync will be bundled and configurable from Synapse's config file or replaced by native support
  * Logs confirming that whatever (proxy or native) support is enabled on the instance and user may now test it
  * Benchmarks and risks are also documented and provided to the homeserver admins so they can decide if want to use it
  * No more "this is actually going to be supported in another release" apologetics

**Comment by avsm** [comment]
I've just upgraded to version 1.7.5 of Element X which was released on the App Store just now for Apple devices, and it's working great! Thank you to for all the hard work that went into this, Matrix devs :-) @erikjohnston

**Comment by paboum** [comment]
I see 6 open bugs with "sliding sync" in their names. So it's probably safe to say that sliding sync feature is now in beta tests phase and `v1.115.0` will not be the final feature release. I would also love learn hw requirements for new instances with this feature (ideally, estimates for 10, 100, 1k, 10k users).

**Comment by spaetz** [comment]
Element X Android 0.6.2 or so enabled simplified sliding sync by default, so this issue should be moot by now and can be closed?

**Comment by wrjlewis** [comment]
Yes this issue can now be closed. Both the Element X Android and iOS public builds have support for simplified sliding sync.

**Comment by paboum** [comment]
Apparently, the docu is still saying the feature support is experimental:
https://github.com/element-hq/synapse/blob/13dea6949bee2820632d4937beee98434bac9a8c/docs/admin_api/experimental_features.md?plain=1#L7
https://github.com/element-hq/synapse/blob/13dea6949bee2820632d4937beee98434bac9a8c/docs/development/synapse_architecture/faster_joins.md?plain=1#L190

**Comment by paboum** [comment]
Funny to repeat myself, another synapse release incoming and another 6 changes regarding sliding sync. I wonder if we are in beta test phase already or not quite there yet. It would really help to publish such information in release notes, we don't pay you so we don't care when it's done, but using an unstable feature on production for unestimated period is not the greates idea. Is it at least safe to use pre-v1.114.0 synapse versions to be sure that none of these bugs affects stability or performance of the instance?

**Comment by wrjlewis** [comment]
We do not consider sliding sync to be in beta nor a test phase, and expect incremental improvements to happen just like we do with Synapse as a whole. On your point re Synapse versions, I would recommend using the latest stable (non-rc) version of Synapse.

**Comment by p-rintz** [comment]
Im not sure where to ask, so I apologize for kinda hijacking this. 

But its still not working for me. I am on Synapse 1.116 and Element X 1.8.4 and its just saying that it cant connect to my homeserver every time. 

Is there any documentation how to successfully get sliding sync working somewhere?

**Comment by paboum** [comment]
> Is there any documentation how to successfully get sliding sync working somewhere?

Not to my knowledge, no. I've had some successful experiments but it was rather cumbersome and mostly trial-and-error, so we could say that's not a mature and documented process. Currently waiting to get the feature stabilised so maybe I try again, until then using SimpleX for fun, as there are no integration issues there.

**Comment by spaetz** [comment]
There is no documentation as there is none needed. Recent synapse servers have it built in and enabled by default. So have the 'X' clients. Support for the separate sync proxy is fading out and to be removed completely. As to that, there is nothing experimental if you are on recent versions.

As to TLS 1.3, I doubt that this is the issue. I run a server on a TLS 1.3 setup just fine. But that might depend on your Android version, so older versions could have problems there.

**Comment by p-rintz** [comment]
> As to TLS 1.3, I doubt that this is the issue. I run a server on a TLS 1.3 setup just fine. But that might depend on your Android version, so older versions could have problems there.

It is TLS1.3 
At least for iOS.
The issue is here: https://github.com/element-hq/element-x-ios/issues/786

**Comment by spaetz** [comment]
Uhh, interesting, thanks. Will need to check if it falls back to TLS 1.2 for me then. Anyway, not part of this issue.

**Comment by paboum** [comment]
https://matrix.org/blog/2024/11/14/moving-to-native-sliding-sync/

Ahh, finally we learn that MSC3575 is basically dead and replaced with MSC4186 which requires no user action, is not yet supported by element-web (which instead includes a confusing message saying "your server lacks support") and is basically in beta until 17.01.2025. If only we knew before, so much waste of time could be avoided.

**Comment by inistor** [comment]
For anyone else who, as myself, did not understand if Simplified SS is *not* supported in Element Web/Desktop (i.e. _not_ X), an answer: https://github.com/element-hq/element-web/issues/28736

My understanding is:
- Synapse supports Simplified SS (since 1.118, I think)
- Element X also supports it, so they work seamlessly - but Element X does support also the "legacy" "Full" SS (MSC3575)
- Element Web/Desktop only supports (as of today) the "legacy" "Full" SS (MSC3575), but, according to the issue above, will support Simplified SS

If I am wrong, please correct me.

**Comment by spaetz** [comment]
Nearly correct, yes:
- synapse supports SSS since 1.114
- legacy sliding sync will be removed from the Element-X clients in mid-January (https://matrix.org/blog/2024/11/14/moving-to-native-sliding-sync/)
